runelite-client/src/main/java/net/runelite/client/plugins/woodcutting/WoodcuttingPlugin.java

```diff
@@ -190,11 +198,31 @@ public void onChatMessage(ChatMessage event)
 
 			if (event.getMessage().contains("A bird's nest falls out of the tree") && config.showNestNotification())
 			{
-				notifier.notify("A bird nest has spawned!");
+				if (clueTierSpawned == null || clueTierSpawned.ordinal() >= config.clueNestNotifyTier().ordinal())
+				{
+					notifier.notify("A bird nest has spawned!");
+				}
+				// Clear the clue tier that has previously spawned
+				clueTierSpawned = null;
 			}
 		}
 	}
 
+	@Subscribe
+	public void onItemSpawned(ItemSpawned itemSpawned)
+	{
+		if (CLUE_NEST_IDS.contains(itemSpawned.getItem().getId()))
```

Comment:
I'm quite wary of maintaining lists of bird nest IDs since these have changed over time--specifically more nests are added every now and then. Instead, because the one consistent feature of bird nests is that they have very consistent item names, I think the entirety of this subscriber would be better written as:

```java
final int spawnedItemId = itemSpawned.getItem().getId();
if ("Bird nest".equals(ItemManager.getItemComposition(spawnedItemId).getName()))
{
	clueTierSpawned = null;
}
else
{
	clueTierSpawned = ClueNestTier.getTierFromItem(spawnedItemId);
}
```

This would remove the need for the `CLUE_NEST_IDS` and `NON_CLUE_NEST_IDS` lists entirely.

Reply:
I understand, and you're right, I haven't given new item release enough thought. Your suggestion might not work without changing `getTierFromItem` to return `null` on getting itemId not in `CLUE_NEST_IDS` however (and this creates a somewhat unnecessary code since checking the item name for the string `"Bird nest"` is redundant).

If I think about what we expect from the code once an update is released adding some new kind of nests or stuff that falls from trees,
1. Not notify the player _only_ if we're _certain_ that a clue nest of low tier has dropped.
2. Still work if an unknown type of nest fell.
With this in mind, we have two options for `onItemSpawned`, tell me what you think is best:
- Live with the rare bug I mentioned in another comment and always reset `clueTierSpawned` like you already suggested
```
if (CLUE_NEST_IDS.contains(itemSpawned.getItem().getId()))
{
	clueTierSpawned = ClueNestTier.getTierFromItem(itemSpawned.getItem().getId());
}
else
{
	clueTierSpawned = null;
}
```
- Try to guess that any new nests will include the specific string "nest" in them (or anything else you think is better)
```
if (CLUE_NEST_IDS.contains(itemSpawned.getItem().getId()))
{
	clueTierSpawned = ClueNestTier.getTierFromItem(itemSpawned.getItem().getId());
}
else if (itemManager.getItemComposition(itemSpawned.getItem().getId()).getName().contains("nest"))
{
	clueTierSpawned = null;
}
```

Reply:
[`Map#get(Object key)` returns `null` if the given key isn't present in the map](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#get-java.lang.Object-), so `getTierFromItem` already behaves the way you described. I'm not really sure what you mean by the "Bird nest" string check being redundant--[clue nests are named "Clue nest (tier)", and all other bird nests are named "Bird nest".](https://chisel.weirdgloop.org/moid/item_name.html#/nest/)

With those two points in mind, my code handles all the following cases:
* Non-clue bird nest is dropped (not by woodcutting): chat message subscriber isn't fired, no notification
* Non-clue bird nest is dropped by woodcutting: `clueTierSpawned` is set to null, chat message subscriber is fired
* Clue nest is dropped by the player: chat message subscriber isn't fired, no notification
* Clue nest is dropped by woodcutting: `clueTierSpawned` is set to the appropriate tier, chat message subscriber is fired
* Non-nest item is dropped: `clueTierSpawned` is set to null (because non-clue nest items aren't in the map), chat message subscriber isn't fired

The only case which could use coverage would be another (or multiple other) items spawning on the same tick as a bird nest spawn. To handle this, I'd change the following:
1. Set `clueTierSpawned` to null on GameTick. This event fires at the end of a tick, after other subscribers execute, so it will not interfere with the other changes in this PR, and simply resets the value after evaluation
2. Skip all item spawn checks if `clueTierSpawned` is not null--other item spawns don't matter if a clue nest has spawned, as they wouldn't impact the value

Reply:
Thanks! Didn't know this information about `get`, good to know. Fixed this as per your suggestion.
Just to clarify on my comment:
```java
final int spawnedItemId = itemSpawned.getItem().getId();
// if ("Bird nest".equals(ItemManager.getItemComposition(spawnedItemId).getName()))
// {
// 	clueTierSpawned = null;
// }
else
{
	clueTierSpawned = ClueNestTier.getTierFromItem(spawnedItemId);
}
``` 
Would be identical to your code since `clueTierSpawned` will be set to null even if the item is a non-clue bird nest since `getTierFromItem` will return `null`